pkg/server/server.go

```diff
@@ -666,40 +674,47 @@ func (s *ProxyServer) Connect(stream agent.AgentService_ConnectServer) error {
 
 	recvCh := make(chan *client.Packet, xfrChannelSize)
 
-	go s.serveRecvBackend(backend, stream, agentID, recvCh)
+	labels := runpprof.Labels(
+		"serverID", s.serverID,
+		"serverCount", strconv.Itoa(s.serverCount),
+		"agentID", agentID,
+	)
```

Comment:
Since this function (`Connect(...)`) can block (the `return <-stopCh` at the end), it might be good to apply these labels to the whole function. I think you can use

```
ctx := runpporf.WithLabels(context.Background(), labels)
runpprof.SetGoroutineLabels(ctx)
```

towards the top of  the function to do that?

Reply:
This is dangerous. SetGoroutineLabels clobbers any previously set labels. So we stand the chance of wiping out labels we were sticky if we do this more generally. Given this method is called be the GRPC framework I think its safe to do in this case. Done.